script/formatting/formatter.py

```diff
@@ -140,6 +142,17 @@ def format_dir(dir_path, update_header, clang_format_code):
     #END FOR [os.walk]
 #END ADD_HEADERS_DIR(DIR_PATH)
 
+#find clang-format executable
+def find_clangformat():
```

Comment:
This is duplicated in source_validator.py. Would it be reasonable to move this into a library or common file and have a single implementation?